### PR TITLE
libnet: add version 1.2

### PR DESCRIPTION
**Pull request by RunThem**
[libnet](https://github.com/libnet/libnet)

### CODE REVIEW

File: packages/l/libnet/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package("libnet")
+    set_homepage("https://codedocs.xyz/libnet/libnet/")
+    set_description("A portable framework for low-level network packet construction")
+
+    add_urls("https://github.com/libnet/libnet/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/libnet/libnet.git")
+    add_versions("v1.2", "b7a371a337d242c017f3471d70bea2963596bec5bd3bd0e33e8517550e2311ef")
+
+    add_deps("autoconf", "automake", "libtool")
```
Comment (waruqi) [inline packages/l/libnet/xmake.lua:9]: tar.gz has configure script, we need not add them

```lua
    on_load(function (package)
        if package:gitref() then
            package:add("deps", "automake", "autoconf", "libtool")
        end
    end)
```

File: packages/l/libnet/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    on_install(function (package)
```
Comment (waruqi) [inline packages/l/libnet/xmake.lua:11]: please disable some unsupported platforms. you can refer other package.

Comment (RunThem) [reply packages/l/libnet/xmake.lua]: This library supports FreeBSD, so what platform name should be filled in.

Comment (waruqi) [reply packages/l/libnet/xmake.lua]: > This library supports FreeBSD, so what platform name should be filled in.

bsd

File: packages/l/libnet/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+             "https://github.com/libnet/libnet.git")
+    add_versions("v1.2", "b7a371a337d242c017f3471d70bea2963596bec5bd3bd0e33e8517550e2311ef")
+
+    on_load(function (package)
+        if package:gitref() then
+            package:add("deps", "automake", "autoconf", "libtool")
+        end
+    end)
+
+    on_install("linux", "macosx", "windows", function (package)
```
Comment (waruqi) [inline packages/l/libnet/xmake.lua:15]: autoconf does not support windows. you can try `mingw`

**Pull request merged**